### PR TITLE
code clean for podgc

### PR DESCRIPTION
**Pull request by andyxning**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
`gcTerminated` is only called when `terminatedPodThreshold` is above zero. So `deleteCount` can only be smaller than `terminatedPodCount `.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
 NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by andyxning** [comment]
/ping @davidopp @enisoc @kow3ns @smarterclayton @dchen1107 @Random-Liu

**Comment by BenTheElder** [comment]
/uncc
this is a lot of assigned people, I'm not currently a reviewer of this code.

**Comment by andyxning** [comment]
Does anyone can help to review this code change? :-(

**Comment by BenTheElder** [comment]
sorry :/
linked it on slack again and CCed you.

**Comment by andyxning** [comment]
Friendly ping.

File: pkg/controller/podgc/gc_controller.go [hunk 1]
```diff
@@ -135,9 +135,6 @@ func (gcc *PodGCController) gcTerminated(pods []*v1.Pod) {
 	terminatedPodCount := len(terminatedPods)
 	deleteCount := terminatedPodCount - gcc.terminatedPodThreshold
 
-	if deleteCount > terminatedPodCount {
```
Comment (liggitt) [inline pkg/controller/podgc/gc_controller.go:138]: removing this does not look correct... terminatedPodThreshold can be < 0:

> fs.Int32Var(&o.TerminatedPodGCThreshold, "terminated-pod-gc-threshold", o.TerminatedPodGCThreshold, "Number of terminated pods that can exist before the terminated pod garbage collector starts deleting terminated pods. If <= 0, the terminated pod garbage collector is disabled.")

**Comment by andyxning** [comment]
@liggitt Big thanks.

**Pull request merged**